### ISSUE TITLE
Select element not re-rendering correctly

### ISSUE THREAD

**Issue by keean** [ionic-team/stencil#1904]
**Stencil version:**

```
 @stencil/core@1.5.3 
```

**I'm submitting a:**

[x ] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
Select element does not seem to update properly. Selecting an option _lower_ down the list works, but when selecting higher up, it does not update. In the example it starts with "Volvo", try selecting "Audi" (works so far) and then "Saab" (Audi us still displayed).

**Expected behavior:**
Select element should always update to selected item 

**Steps to reproduce:**

Build a simple html page containing the component below, and then try selecting different options.

**Related code:**

```tsx
import {Component, State, h} from '@stencil/core';
  
@Component({
    tag: 'test-select',
    shadow: true
})
export class TestSelect {
    @State() selectValue: string;

    private handleSelect(event) {
        this.selectValue = event.target.value;
    }

    render() {
        return <select onInput={event => this.handleSelect(event)}>
            <option value="volvo" selected={this.selectValue === 'volvo'}>Volvo</option>
            <option value="saab" selected={this.selectValue === 'saab'}>Saab</option>
            <option value="mercedes" selected={this.selectValue === 'mercedes'}>Mercedes</option>
            <option value="audi" selected={this.selectValue === 'audi'}>Audi</option>
        </select>
    }
}
```

**Other information:**

This was tested on Safari, Firefox nor Chrome on Mac OS Mojave (10.14.6)

**Comment by manucorporat** [comment]
Should be fixed in @stencil/core 1.5.4!

**Comment by keean** [comment]
Great, tested and it works. Thanks.
